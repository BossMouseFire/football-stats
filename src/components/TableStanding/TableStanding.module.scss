.tableStandings {
  position: relative;
  table {
    display: inline-block;
    border-collapse: collapse;
    text-align: center;
    max-height: 600px;
    overflow-y: scroll;

    .head {
      th {
        background-color: #608D2E;
        color: white;
      }
      position: sticky;
      top: 0;
      left: 0;
    }

    td{
      div{
        display: flex;
        flex-direction: row;
        align-items: center;
        img{
          width: 30px;
        }
        span{
          margin-left: 20px;
          cursor: pointer;
          transition: color 500ms ease;
          &:hover{
            color: #1B5500 !important;
            transition: color 500ms ease;
          }
        }
      }
    }
    tr{
      &:nth-of-type(2),
      &:nth-of-type(3),
      &:nth-of-type(4)
      {
        td:nth-of-type(1){
          color: #3DB903;
        }
      }
    }
    th, td {
      padding: 15px 20px;
    }
  }
}
