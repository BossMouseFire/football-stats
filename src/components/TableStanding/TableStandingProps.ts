import { TableTeamStat } from '../../model/TableTeamStat';

export interface TableStandingProps {
    table: TableTeamStat[];
}

export interface TeamStandingProps {
    teamStat: TableTeamStat;
}
