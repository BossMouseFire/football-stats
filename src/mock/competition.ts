export const competitionMock = {
    area: {
        id: 2224,
        name: 'Spain',
        code: 'ESP',
        flag: 'https://crests.football-data.org/760.svg',
    },
    id: 2014,
    name: 'Primera Division',
    code: 'PD',
    type: 'LEAGUE',
    emblem: 'https://crests.football-data.org/PD.png',
    currentSeason: {
        id: 1577,
        startDate: '2023-08-13',
        endDate: '2024-05-26',
        currentMatchday: 13,
        winner: null,
    },
    lastUpdated: '2022-03-20T09:20:08Z',
};

export const competitionMockWithWinner = {
    area: {
        id: 2267,
        name: 'World',
        code: 'INT',
        flag: null,
    },
    id: 2000,
    name: 'FIFA World Cup',
    code: 'WC',
    type: 'CUP',
    emblem: 'https://crests.football-data.org/qatar.png',
    currentSeason: {
        id: 1382,
        startDate: '2022-11-20',
        endDate: '2022-12-18',
        currentMatchday: 8,
        winner: {
            id: 762,
            name: 'Argentina',
            shortName: 'Argentina',
            tla: 'ARG',
            crest: 'https://crests.football-data.org/762.png',
            address: 'Viamonte 1366/76 Buenos Aires, Buenos Aires 1053',
            website: 'http://www.afa.org.ar',
            founded: 1893,
            clubColors: 'Sky Blue / White / Black',
            venue: null,
            lastUpdated: '2022-05-17T21:09:25Z',
        },
    },
    lastUpdated: '2022-05-09T19:45:29Z',
};
