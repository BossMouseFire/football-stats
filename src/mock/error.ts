import { ErrorData } from '../errors/ErrorData';

export const errorMock404 = {
    message: 'The resource you are looking for does not exist.',
    error: 404,
};

export const errorMock400 = new ErrorData({
    message:
        "Argument 'competitionId' is expected to be either an integer in a specified range or a competition code.",
    errorCode: 400,
});

export const errorMock403 = new ErrorData({
    message:
        'The resource you are looking for is restricted and apparently not within your permissions. Please check your subscription.',
    errorCode: 403,
});
