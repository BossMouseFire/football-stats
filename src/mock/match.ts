import { MatchDto } from '../dto/MatchDto';

export const matchesMock: MatchDto[] = [
    {
        season: {
            id: 1577,
            startDate: '2023-08-13',
            endDate: '2024-05-26',
            currentMatchday: 13,
            winner: null,
        },
        id: 438482,
        utcDate: '2023-08-11T17:30:00Z',
        status: 'FINISHED',
        homeTeam: {
            id: 267,
            name: 'UD Almería',
            shortName: 'Almería',
            tla: 'ALM',
            crest: 'https://crests.football-data.org/267.png',
        },
        awayTeam: {
            id: 87,
            name: 'Rayo Vallecano de Madrid',
            shortName: 'Rayo Vallecano',
            tla: 'RAY',
            crest: 'https://crests.football-data.org/87.svg',
        },
        score: {
            winner: 'AWAY_TEAM',
            duration: 'REGULAR',
            fullTime: {
                home: 0,
                away: 2,
            },
            halfTime: {
                home: 0,
                away: 2,
            },
        },
    },
    {
        season: {
            id: 1577,
            startDate: '2023-08-13',
            endDate: '2024-05-26',
            currentMatchday: 13,
            winner: null,
        },
        id: 438479,
        utcDate: '2023-08-11T20:00:00Z',
        status: 'FINISHED',
        homeTeam: {
            id: 559,
            name: 'Sevilla FC',
            shortName: 'Sevilla FC',
            tla: 'SEV',
            crest: 'https://crests.football-data.org/559.svg',
        },
        awayTeam: {
            id: 95,
            name: 'Valencia CF',
            shortName: 'Valencia',
            tla: 'VAL',
            crest: 'https://crests.football-data.org/95.svg',
        },
        score: {
            winner: 'AWAY_TEAM',
            duration: 'REGULAR',
            fullTime: {
                home: 1,
                away: 2,
            },
            halfTime: {
                home: 0,
                away: 0,
            },
        },
    },
];
