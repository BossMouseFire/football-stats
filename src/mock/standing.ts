import { Standing } from '../model/Standing';
import { Competition } from '../model/Competition';
import { StandingDto } from '../dto/StandingDto';
import { CompetitionDto } from '../dto/CompetitionDto';
import { SeasonDto } from '../dto/SeasonDto';

type StandingCompetition = Pick<
    CompetitionDto,
    'id' | 'name' | 'emblem' | 'area' | 'code' | 'type'
>;

export const standingsMock: {
    standings: StandingDto[];
    competition: StandingCompetition;
    season: SeasonDto;
} = {
    competition: {
        id: 2015,
        name: 'Ligue 1',
        code: 'FL1',
        type: 'LEAGUE',
        emblem: 'https://crests.football-data.org/FL1.png',
    },
    season: {
        id: 1595,
        startDate: '2023-08-13',
        endDate: '2024-05-18',
        currentMatchday: 12,
        winner: null,
    },
    standings: [
        {
            stage: 'REGULAR_SEASON',
            type: 'TOTAL',
            table: [
                {
                    position: 1,
                    team: {
                        id: 524,
                        name: 'Paris Saint-Germain FC',
                        shortName: 'PSG',
                        tla: 'PSG',
                        crest: 'https://crests.football-data.org/524.png',
                    },
                    playedGames: 12,
                    form: null,
                    won: 8,
                    draw: 3,
                    lost: 1,
                    points: 27,
                    goalsFor: 29,
                    goalsAgainst: 9,
                    goalDifference: 20,
                },
                {
                    position: 2,
                    team: {
                        id: 522,
                        name: 'OGC Nice',
                        shortName: 'Nice',
                        tla: 'NIC',
                        crest: 'https://crests.football-data.org/522.png',
                    },
                    playedGames: 12,
                    form: null,
                    won: 7,
                    draw: 5,
                    lost: 0,
                    points: 26,
                    goalsFor: 13,
                    goalsAgainst: 4,
                    goalDifference: 9,
                },
                {
                    position: 3,
                    team: {
                        id: 548,
                        name: 'AS Monaco FC',
                        shortName: 'Monaco',
                        tla: 'ASM',
                        crest: 'https://crests.football-data.org/548.png',
                    },
                    playedGames: 12,
                    form: null,
                    won: 7,
                    draw: 3,
                    lost: 2,
                    points: 24,
                    goalsFor: 25,
                    goalsAgainst: 14,
                    goalDifference: 11,
                },
                {
                    position: 4,
                    team: {
                        id: 521,
                        name: 'Lille OSC',
                        shortName: 'Lille',
                        tla: 'LIL',
                        crest: 'https://crests.football-data.org/521.svg',
                    },
                    playedGames: 12,
                    form: null,
                    won: 5,
                    draw: 5,
                    lost: 2,
                    points: 20,
                    goalsFor: 15,
                    goalsAgainst: 11,
                    goalDifference: 4,
                },
                {
                    position: 5,
                    team: {
                        id: 547,
                        name: 'Stade de Reims',
                        shortName: 'Stade de Reims',
                        tla: 'SDR',
                        crest: 'https://crests.football-data.org/547.png',
                    },
                    playedGames: 12,
                    form: null,
                    won: 6,
                    draw: 2,
                    lost: 4,
                    points: 20,
                    goalsFor: 17,
                    goalsAgainst: 15,
                    goalDifference: 2,
                },
                {
                    position: 6,
                    team: {
                        id: 533,
                        name: 'Le Havre AC',
                        shortName: 'Le Havre',
                        tla: 'HAC',
                        crest: 'https://crests.football-data.org/533.png',
                    },
                    playedGames: 12,
                    form: null,
                    won: 3,
                    draw: 6,
                    lost: 3,
                    points: 15,
                    goalsFor: 12,
                    goalsAgainst: 13,
                    goalDifference: -1,
                },
                {
                    position: 7,
                    team: {
                        id: 512,
                        name: 'Stade Brestois 29',
                        shortName: 'Brest',
                        tla: 'BRE',
                        crest: 'https://crests.football-data.org/512.png',
                    },
                    playedGames: 11,
                    form: null,
                    won: 4,
                    draw: 3,
                    lost: 4,
                    points: 15,
                    goalsFor: 11,
                    goalsAgainst: 13,
                    goalDifference: -2,
                },
                {
                    position: 8,
                    team: {
                        id: 543,
                        name: 'FC Nantes',
                        shortName: 'Nantes',
                        tla: 'NAN',
                        crest: 'https://crests.football-data.org/543.png',
                    },
                    playedGames: 12,
                    form: null,
                    won: 4,
                    draw: 2,
                    lost: 6,
                    points: 14,
                    goalsFor: 17,
                    goalsAgainst: 23,
                    goalDifference: -6,
                },
                {
                    position: 9,
                    team: {
                        id: 516,
                        name: 'Olympique de Marseille',
                        shortName: 'Marseille',
                        tla: 'MAR',
                        crest: 'https://crests.football-data.org/516.png',
                    },
                    playedGames: 10,
                    form: null,
                    won: 3,
                    draw: 4,
                    lost: 3,
                    points: 13,
                    goalsFor: 12,
                    goalsAgainst: 12,
                    goalDifference: 0,
                },
                {
                    position: 10,
                    team: {
                        id: 546,
                        name: 'Racing Club de Lens',
                        shortName: 'RC Lens',
                        tla: 'RCL',
                        crest: 'https://crests.football-data.org/546.png',
                    },
                    playedGames: 11,
                    form: null,
                    won: 3,
                    draw: 4,
                    lost: 4,
                    points: 13,
                    goalsFor: 12,
                    goalsAgainst: 13,
                    goalDifference: -1,
                },
                {
                    position: 11,
                    team: {
                        id: 545,
                        name: 'FC Metz',
                        shortName: 'FC Metz',
                        tla: 'FCM',
                        crest: 'https://crests.football-data.org/545.svg',
                    },
                    playedGames: 12,
                    form: null,
                    won: 3,
                    draw: 4,
                    lost: 5,
                    points: 13,
                    goalsFor: 12,
                    goalsAgainst: 18,
                    goalDifference: -6,
                },
                {
                    position: 12,
                    team: {
                        id: 518,
                        name: 'Montpellier HSC',
                        shortName: 'Montpellier',
                        tla: 'MON',
                        crest: 'https://crests.football-data.org/518.png',
                    },
                    playedGames: 11,
                    form: null,
                    won: 3,
                    draw: 4,
                    lost: 4,
                    points: 12,
                    goalsFor: 15,
                    goalsAgainst: 14,
                    goalDifference: 1,
                },
                {
                    position: 13,
                    team: {
                        id: 529,
                        name: 'Stade Rennais FC 1901',
                        shortName: 'Stade Rennais',
                        tla: 'REN',
                        crest: 'https://crests.football-data.org/529.png',
                    },
                    playedGames: 12,
                    form: null,
                    won: 2,
                    draw: 6,
                    lost: 4,
                    points: 12,
                    goalsFor: 16,
                    goalsAgainst: 16,
                    goalDifference: 0,
                },
                {
                    position: 14,
                    team: {
                        id: 511,
                        name: 'Toulouse FC',
                        shortName: 'Toulouse',
                        tla: 'TOU',
                        crest: 'https://crests.football-data.org/511.png',
                    },
                    playedGames: 12,
                    form: null,
                    won: 2,
                    draw: 6,
                    lost: 4,
                    points: 12,
                    goalsFor: 13,
                    goalsAgainst: 16,
                    goalDifference: -3,
                },
                {
                    position: 15,
                    team: {
                        id: 576,
                        name: 'RC Strasbourg Alsace',
                        shortName: 'Strasbourg',
                        tla: 'RC ',
                        crest: 'https://crests.football-data.org/576.png',
                    },
                    playedGames: 11,
                    form: null,
                    won: 3,
                    draw: 3,
                    lost: 5,
                    points: 12,
                    goalsFor: 9,
                    goalsAgainst: 15,
                    goalDifference: -6,
                },
                {
                    position: 16,
                    team: {
                        id: 525,
                        name: 'FC Lorient',
                        shortName: 'Lorient',
                        tla: 'FCL',
                        crest: 'https://crests.football-data.org/525.png',
                    },
                    playedGames: 12,
                    form: null,
                    won: 2,
                    draw: 5,
                    lost: 5,
                    points: 11,
                    goalsFor: 15,
                    goalsAgainst: 21,
                    goalDifference: -6,
                },
                {
                    position: 17,
                    team: {
                        id: 541,
                        name: 'Clermont Foot 63',
                        shortName: 'Clermont Foot',
                        tla: 'CLF',
                        crest: 'https://crests.football-data.org/541.svg',
                    },
                    playedGames: 11,
                    form: null,
                    won: 2,
                    draw: 3,
                    lost: 6,
                    points: 9,
                    goalsFor: 8,
                    goalsAgainst: 14,
                    goalDifference: -6,
                },
                {
                    position: 18,
                    team: {
                        id: 523,
                        name: 'Olympique Lyonnais',
                        shortName: 'Olympique Lyon',
                        tla: 'LYO',
                        crest: 'https://crests.football-data.org/523.svg',
                    },
                    playedGames: 11,
                    form: null,
                    won: 1,
                    draw: 4,
                    lost: 6,
                    points: 7,
                    goalsFor: 9,
                    goalsAgainst: 19,
                    goalDifference: -10,
                },
            ],
        },
    ],
};

export const standingsMockRefactor = {
    standings: standingsMock.standings.map((item) => new Standing(item)),
    competition: new Competition({
        ...standingsMock.competition,
        currentSeason: standingsMock.season,
    }),
};
