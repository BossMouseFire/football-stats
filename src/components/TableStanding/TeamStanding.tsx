import React from 'react';
import { TeamStandingProps } from './TableStandingProps';

export const TeamStanding: React.FC<TeamStandingProps> = ({ teamStat }) => {
    return (
        <tr>
            <td>{teamStat.position}</td>
            <td>
                <div>
                    <img src={teamStat.team.crest} />
                    <span>{teamStat.team.name}</span>
                </div>
            </td>
            <td>{teamStat.playedGames}</td>
            <td>{teamStat.won}</td>
            <td>{teamStat.draw}</td>
            <td>{teamStat.lost}</td>
            <td>{teamStat.goalDifference}</td>
            <td>{teamStat.points}</td>
        </tr>
    );
};
