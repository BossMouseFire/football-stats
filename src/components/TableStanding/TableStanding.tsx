import React from 'react';
import { TeamStanding } from './TeamStanding';
import { TableStandingProps } from './TableStandingProps';

import styles from './TableStanding.module.scss';

export const TableStanding: React.FC<TableStandingProps> = ({ table }) => {
    return (
        <div className={styles.tableStandings}>
            <table>
                <tr className={styles.head}>
                    <th>#</th>
                    <th>Команда</th>
                    <th>С</th>
                    <th>В</th>
                    <th>Н</th>
                    <th>П</th>
                    <th>РГ</th>
                    <th>Очки</th>
                </tr>
                {table.map((team) => (
                    <TeamStanding teamStat={team} key={team?.team.id} />
                ))}
            </table>
        </div>
    );
};
