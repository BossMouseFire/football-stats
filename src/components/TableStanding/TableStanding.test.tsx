import { TableStanding } from './TableStanding';
import { renderToJson } from '../../test/helpers/renderToJson';
import { standingsMock } from '../../mock/standing';

it('Table standing renders correctly', () => {
    const table = renderToJson(
        <TableStanding table={standingsMock.standings[0].table} />,
    );

    expect(table).toMatchSnapshot();
});

export {};
