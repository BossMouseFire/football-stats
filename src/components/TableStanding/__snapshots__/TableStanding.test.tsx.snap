// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Table standing renders correctly 1`] = `
<div
  className="tableStandings"
>
  <table>
    <tr
      className="head"
    >
      <th>
        #
      </th>
      <th>
        Команда
      </th>
      <th>
        С
      </th>
      <th>
        В
      </th>
      <th>
        Н
      </th>
      <th>
        П
      </th>
      <th>
        РГ
      </th>
      <th>
        Очки
      </th>
    </tr>
    <tr>
      <td>
        1
      </td>
      <td>
        <div>
          <img
            src="https://crests.football-data.org/524.png"
          />
          <span>
            Paris Saint-Germain FC
          </span>
        </div>
      </td>
      <td>
        12
      </td>
      <td>
        8
      </td>
      <td>
        3
      </td>
      <td>
        1
      </td>
      <td>
        20
      </td>
      <td>
        27
      </td>
    </tr>
    <tr>
      <td>
        2
      </td>
      <td>
        <div>
          <img
            src="https://crests.football-data.org/522.png"
          />
          <span>
            OGC Nice
          </span>
        </div>
      </td>
      <td>
        12
      </td>
      <td>
        7
      </td>
      <td>
        5
      </td>
      <td>
        0
      </td>
      <td>
        9
      </td>
      <td>
        26
      </td>
    </tr>
    <tr>
      <td>
        3
      </td>
      <td>
        <div>
          <img
            src="https://crests.football-data.org/548.png"
          />
          <span>
            AS Monaco FC
          </span>
        </div>
      </td>
      <td>
        12
      </td>
      <td>
        7
      </td>
      <td>
        3
      </td>
      <td>
        2
      </td>
      <td>
        11
      </td>
      <td>
        24
      </td>
    </tr>
    <tr>
      <td>
        4
      </td>
      <td>
        <div>
          <img
            src="https://crests.football-data.org/521.svg"
          />
          <span>
            Lille OSC
          </span>
        </div>
      </td>
      <td>
        12
      </td>
      <td>
        5
      </td>
      <td>
        5
      </td>
      <td>
        2
      </td>
      <td>
        4
      </td>
      <td>
        20
      </td>
    </tr>
    <tr>
      <td>
        5
      </td>
      <td>
        <div>
          <img
            src="https://crests.football-data.org/547.png"
          />
          <span>
            Stade de Reims
          </span>
        </div>
      </td>
      <td>
        12
      </td>
      <td>
        6
      </td>
      <td>
        2
      </td>
      <td>
        4
      </td>
      <td>
        2
      </td>
      <td>
        20
      </td>
    </tr>
    <tr>
      <td>
        6
      </td>
      <td>
        <div>
          <img
            src="https://crests.football-data.org/533.png"
          />
          <span>
            Le Havre AC
          </span>
        </div>
      </td>
      <td>
        12
      </td>
      <td>
        3
      </td>
      <td>
        6
      </td>
      <td>
        3
      </td>
      <td>
        -1
      </td>
      <td>
        15
      </td>
    </tr>
    <tr>
      <td>
        7
      </td>
      <td>
        <div>
          <img
            src="https://crests.football-data.org/512.png"
          />
          <span>
            Stade Brestois 29
          </span>
        </div>
      </td>
      <td>
        11
      </td>
      <td>
        4
      </td>
      <td>
        3
      </td>
      <td>
        4
      </td>
      <td>
        -2
      </td>
      <td>
        15
      </td>
    </tr>
    <tr>
      <td>
        8
      </td>
      <td>
        <div>
          <img
            src="https://crests.football-data.org/543.png"
          />
          <span>
            FC Nantes
          </span>
        </div>
      </td>
      <td>
        12
      </td>
      <td>
        4
      </td>
      <td>
        2
      </td>
      <td>
        6
      </td>
      <td>
        -6
      </td>
      <td>
        14
      </td>
    </tr>
    <tr>
      <td>
        9
      </td>
      <td>
        <div>
          <img
            src="https://crests.football-data.org/516.png"
          />
          <span>
            Olympique de Marseille
          </span>
        </div>
      </td>
      <td>
        10
      </td>
      <td>
        3
      </td>
      <td>
        4
      </td>
      <td>
        3
      </td>
      <td>
        0
      </td>
      <td>
        13
      </td>
    </tr>
    <tr>
      <td>
        10
      </td>
      <td>
        <div>
          <img
            src="https://crests.football-data.org/546.png"
          />
          <span>
            Racing Club de Lens
          </span>
        </div>
      </td>
      <td>
        11
      </td>
      <td>
        3
      </td>
      <td>
        4
      </td>
      <td>
        4
      </td>
      <td>
        -1
      </td>
      <td>
        13
      </td>
    </tr>
    <tr>
      <td>
        11
      </td>
      <td>
        <div>
          <img
            src="https://crests.football-data.org/545.svg"
          />
          <span>
            FC Metz
          </span>
        </div>
      </td>
      <td>
        12
      </td>
      <td>
        3
      </td>
      <td>
        4
      </td>
      <td>
        5
      </td>
      <td>
        -6
      </td>
      <td>
        13
      </td>
    </tr>
    <tr>
      <td>
        12
      </td>
      <td>
        <div>
          <img
            src="https://crests.football-data.org/518.png"
          />
          <span>
            Montpellier HSC
          </span>
        </div>
      </td>
      <td>
        11
      </td>
      <td>
        3
      </td>
      <td>
        4
      </td>
      <td>
        4
      </td>
      <td>
        1
      </td>
      <td>
        12
      </td>
    </tr>
    <tr>
      <td>
        13
      </td>
      <td>
        <div>
          <img
            src="https://crests.football-data.org/529.png"
          />
          <span>
            Stade Rennais FC 1901
          </span>
        </div>
      </td>
      <td>
        12
      </td>
      <td>
        2
      </td>
      <td>
        6
      </td>
      <td>
        4
      </td>
      <td>
        0
      </td>
      <td>
        12
      </td>
    </tr>
    <tr>
      <td>
        14
      </td>
      <td>
        <div>
          <img
            src="https://crests.football-data.org/511.png"
          />
          <span>
            Toulouse FC
          </span>
        </div>
      </td>
      <td>
        12
      </td>
      <td>
        2
      </td>
      <td>
        6
      </td>
      <td>
        4
      </td>
      <td>
        -3
      </td>
      <td>
        12
      </td>
    </tr>
    <tr>
      <td>
        15
      </td>
      <td>
        <div>
          <img
            src="https://crests.football-data.org/576.png"
          />
          <span>
            RC Strasbourg Alsace
          </span>
        </div>
      </td>
      <td>
        11
      </td>
      <td>
        3
      </td>
      <td>
        3
      </td>
      <td>
        5
      </td>
      <td>
        -6
      </td>
      <td>
        12
      </td>
    </tr>
    <tr>
      <td>
        16
      </td>
      <td>
        <div>
          <img
            src="https://crests.football-data.org/525.png"
          />
          <span>
            FC Lorient
          </span>
        </div>
      </td>
      <td>
        12
      </td>
      <td>
        2
      </td>
      <td>
        5
      </td>
      <td>
        5
      </td>
      <td>
        -6
      </td>
      <td>
        11
      </td>
    </tr>
    <tr>
      <td>
        17
      </td>
      <td>
        <div>
          <img
            src="https://crests.football-data.org/541.svg"
          />
          <span>
            Clermont Foot 63
          </span>
        </div>
      </td>
      <td>
        11
      </td>
      <td>
        2
      </td>
      <td>
        3
      </td>
      <td>
        6
      </td>
      <td>
        -6
      </td>
      <td>
        9
      </td>
    </tr>
    <tr>
      <td>
        18
      </td>
      <td>
        <div>
          <img
            src="https://crests.football-data.org/523.svg"
          />
          <span>
            Olympique Lyonnais
          </span>
        </div>
      </td>
      <td>
        11
      </td>
      <td>
        1
      </td>
      <td>
        4
      </td>
      <td>
        6
      </td>
      <td>
        -10
      </td>
      <td>
        7
      </td>
    </tr>
  </table>
</div>
`;
